src/d3d12/d3d12_resource_pool_texture.cpp

```diff
@@ -637,13 +637,82 @@ namespace wr
 
 	}
 
-	TextureHandle D3D12TexturePool::LoadFromRawMemory(char* data, size_t width, size_t height, bool srgb, bool generate_mips)
+	TextureHandle D3D12TexturePool::LoadFromRawMemory(unsigned char* data, size_t width, size_t height, bool srgb, bool generate_mips)
 	{
-		TextureHandle handle;
+		auto device = m_render_system.m_device;
+		DirectX::ScratchImage* scratch_image = new DirectX::ScratchImage;
+		DirectX::Image image = {};
+
+		image.format = srgb ? DXGI_FORMAT_R8G8B8A8_UNORM_SRGB : DXGI_FORMAT_R8G8B8A8_UNORM;
+		image.width = width;
+		image.height = height;
+		image.rowPitch = width * 4;
+		image.slicePitch = width * height * 4;
+		image.pixels = data;
+
+		HRESULT hr = scratch_image->InitializeFromImage(image);
+
+		if (FAILED(hr))
+		{
+			LOGE("Loading raw texture failed with code " + std::to_string(hr));
+		}
 
-		LOGC("THIS FUNCTION IS NOT IMPLEMENTED");
+		uint32_t mip_lvls;
```

Comment:
Maybe, but there are enough differences that you can't really turn them into a single function.

Reply:
I disagree, This can easilly be stripped in either multiple functions, or slimmed down functions. Move code into functions, call loadfrommemory in loadfromfile. see what works best.

Reply:
Fixed